scrapy/extensions/feedexport.py

```diff
@@ -79,7 +80,10 @@ def store(self, file):
 class FileFeedStorage(object):
 
     def __init__(self, uri):
-        self.path = file_uri_to_path(uri)
+        if isinstance(uri, Path):
+            self.path = uri.resolve()
+        else:
+            self.path = file_uri_to_path(uri)
```

Comment:
I think this would be easier to fix on line 191, where `settings['FEED_URI']` is read. I suspect casting settings['FEED_URI'] to a string if it is not `None` should be enough.